plugins/tours/tests/phpunit/unit/registerCustomConfigs.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Tests for register_custom_configs().
+ *
+ * @since       1.0.0
+ * @author      Robert Gadon <rgadon107>
+ * @package     spiralWebDb\CornerstoneTours\Tests\Integration
+ * @link        https://github.com/rgadon107/cornerstone
+ * @license     GNU-2.0+
+ */
+
+namespace spiralWebDb\CornerstoneTours\Tests\Unit;
+
+use Brain\Monkey\Functions;
+use spiralWebDb\Cornerstone\Tests\Unit\Test_Case;
+use function spiralWebDb\CornerstoneTours\register_custom_configs;
+
+/**
+ * @covers ::\spiralWebDb\CornerstoneTours\register_custom_configs
+ *
+ * @group   tours
+ */
+class Tests_RegisterCustomConfigs extends Test_Case {
+
+	/**
+	 * Prepares the test environment before each test.
+	 */
+	protected function setUp() {
+		parent::setUp();
+
+		require_once TOURS_ROOT_DIR . '/src/config-loader.php';
+	}
+
+	/**
+	 * @dataProvider addTestData
+	 */
+	public function test_should_add_runtime_post_type_config_to_existing_configs( array $configurations, array $tours ) {
+		Functions\expect( '_get_plugin_directory' )
+			->times( 1 )
+			->with()
+			->andReturn( TOURS_ROOT_DIR );
+
+		$this->assertArrayHasKey( 'tours', register_custom_configs( (array) $configurations ) );
+		$this->assertContains( $tours['post_type'], (array) $tours );
+		$this->assertContains( $tours['features'], (array) $tours );
+		$this->assertContains( $tours['args'], (array) $tours );
```

Comment:
Look at the `assertContains()`. They are checking the tours array from the test data and _not_ what gets returned from `register_custom_configs()`. How would you fix that?

Reply:
@hellofromtonya You make a good point. I've made multiple attempts to confirm the subarray keys returned by the function `register_custom_configs`. I'm just not able to do it with the programming tools currently available to me in PHPUnit.

I'm attempting to drill down 2 levels into an array to confirm that one or subarray keys are part of the larger configuration array. Not sure that it's even worth proving that those subarray keys exist in the expected array. 

**A second issue:** As in the integration test, is it possible to test a second scenario in the unit test with the data provider where the existing configuration passed to `register_custom_configs()` is not an empty array, but a second, nonempty array added into the `tours` config? I can't seem to set it up properly.